docs/message.md

```diff
@@ -12,8 +15,15 @@ these details can be separated from the actual requests.
 $ composer require php-http/message
 ```
 
+## Authentication
```

Comment:
i would split the message documentation into one page per aspect: auth, stream, decorator, factory, cookie. those docs are not very closely related, and will grow over time.